server/graphql/v2/query/NotificationsQuery.ts

```diff
@@ -0,0 +1,50 @@
+import { GraphQLList, GraphQLNonNull } from 'graphql';
+
+import models from '../../../models';
+import { Forbidden } from '../../errors';
+import { idDecode, IDENTIFIER_TYPES } from '../identifiers';
+import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
+import { Notification } from '../object/Notification';
+
+const NotificationsQuery = {
+  type: new GraphQLList(Notification),
+  description: '[AUTHENTICATED] Returns user notifications',
+  args: {
+    individual: {
+      type: new GraphQLNonNull(AccountReferenceInput),
+      description: 'A reference to an account (usually Individual). Will return notifications set by this user.',
+    },
+    account: {
+      type: AccountReferenceInput,
+      description:
+        'A reference to an account (usually Collective, Fund or Organization). Will return notifications related to specific account.',
+    },
+  },
+  async resolve(_, args, req) {
+    if (!req.remoteUser) {
+      throw new Forbidden('You need to be logged in to query for notifications');
+    }
+
+    const userCollective = await fetchAccountWithReference(args.individual, { throwIfMissing: true });
+    const user = await userCollective.getUser();
+    if (!user) {
+      throw new Error('Individual must be an User account');
+    } else if (user.id !== req.remoteUser.id) {
+      throw new Forbidden('You can only query for your own Notifications');
+    }
```

Comment:
:ice_cream: I would omit the `individual` argument altogether if this is only for currently logged-in user. Or eventually put it on `loggedInUser.notifications` rather than a root query. But that's only a personal preference.

Reply:
Root query doesn't seem really useful indeed. Having a property on `individual` should be good.